ra2ce/graph/networks_utils.py

```diff
@@ -1131,8 +1132,154 @@ def simplify_graph_count(complex_graph: nx.Graph) -> nx.Graph:
+                    # value list
+                    path_attributes[attr].append(edge_data[attr])
+                else:
+                    # if this key doesn't already exist, set the value to a list
+                    # containing the one value
+                    path_attributes[attr] = [edge_data[attr]]
+
+        # consolidate the path's edge segments' attribute values
+        for attr in path_attributes:
+            if attr in attrs_to_sum:
+                # if this attribute must be summed, sum it now
+                path_attributes[attr] = sum(path_attributes[attr])
+            elif attr != 'geometry' and len(set(path_attributes[attr])) == 1:
+                # if there's only 1 unique value in this attribute list,
+                # consolidate it to the single value (the zero-th):
+                path_attributes[attr] = path_attributes[attr][0]
+            elif attr != 'geometry':
+                # otherwise, if there are multiple values, keep one of each
+                path_attributes[attr] = list(set(path_attributes[attr]))
+
+        # construct the new consolidated edge's geometry for this path
+        if all("geometry" in G.nodes[node] for node in path):
+            # Check if all nodes in the path have "geometry" attribute
+            # Use existing geometries to create LineString
+            path_attributes["geometry"] = LineString([G.nodes[node]["geometry"] for node in path])
+        elif all("x" in G.nodes[node] and "y" in G.nodes[node] for node in path):
+            # Create LineString using x and y coordinates
+            path_attributes["geometry"] = LineString(
+                [Point((G.nodes[node]["x"], G.nodes[node]["y"])) for node in path]
+            )
+        else:
+            raise ValueError("All nodes in the path must have 'geometry' or 'x' and 'y' attributes.")
+
+        if track_merged:
+            # add the merged edges as a new attribute of the simplified edge
+            path_attributes["merged_edges"] = merged_edges
+
+        # add the nodes and edge to their lists for processing at the end
+        all_nodes_to_remove.extend(path[1:-1])
+        all_edges_to_add.append(
+            {"origin": path[0], "destination": path[-1], "attr_dict": path_attributes}
+        )
+
+    # for each edge to add in the list we assembled, create a new edge between
+    # the origin and destination
+    for edge in all_edges_to_add:
+        G.add_edge(edge["origin"], edge["destination"], **edge["attr_dict"])
+
+    # finally remove all the interstitial nodes between the new edges
+    G.remove_nodes_from(set(all_nodes_to_remove))
+
+    if remove_rings:
+        G = _remove_rings(G)
+
+    # mark the graph as having been simplified
+    G.graph["simplified"] = True
+
+    msg = (
+        f"Simplified graph: {initial_node_count:,} to {len(G):,} nodes, "
+        f"{initial_edge_count:,} to {len(G.edges):,} edges"
+    )
+    utils.log(msg)
```

Comment:
why using this instead of our own `logging.info` from the default library `logging` ?